src/webapp/run_rebuild.sh

```diff
@@ -1,92 +1,150 @@
+    swap_total=$((swap_total / 1024))
+    swap_free=$((swap_free / 1024))
+    total_free=$((total_free / 1024))
+
+    local free_to_use=$((total_free - mem_buffer))
+
+    if [ "$VERBOSE" == true ]; then
+        echo "  Total phys memory : ${mem_total} MB"
+        echo "  Free phys memory  : ${mem_free} MB"
+        echo "  Total swap memory : ${swap_total} MB"
+        echo "  Free swap memory  : ${swap_free} MB"
+        echo "  Free total memory : ${total_free} MB"
+        echo "  Keep as buffer    : ${mem_buffer} MB"
+        echo -e "  Free usable memory: ${free_to_use} MB\n"
+    fi
+
+    if [[ -z $NODEMEM ]]; then
+        # mininum memory used for node
+        local mem_min=512
+        if [[ $free_to_use -gt $mem_min ]]; then
+            NODEMEM=$free_to_use
+        else
+            echo "  WARN: Not enough memory left on system for node (usable ${free_to_use} MB, min. ${mem_min} MB)."
+            echo "        Trying to adjust swap size ..."
+
+            local add_swap_size=$((mem_min / 2))
+            local new_swap_size=$((swap_total + add_swap_size))
+
+            # keep a buffer on the filesystem
+            local filesystem_needed=$((add_swap_size + 512))
+            local filesystem_free=$(df -BM -P / | tail -n 1 | awk '{print $4}')
+            filesystem_free=${filesystem_free//M}
+
+            if [ "$VERBOSE" == true ]; then
+                echo "  New swap size = $new_swap_size MB"
+                echo "  Additional filesystem space needed = $filesystem_needed MB"
+                echo "  Current free filesystem space = $filesystem_free MB"
+            fi
+
+            if [ "${filesystem_free}" -lt "${filesystem_needed}" ]; then
+                echo "  ERROR: Not enough space available on filesystem for swap (free ${filesystem_free} MB, min. ${filesystem_needed} MB). Abort!"
+                exit 1
+            elif ! change_swap $new_swap_size ; then
+                echo "  ERROR: failed to change swap size. Abort!"
+                exit 1
+            fi
+
+            calc_nodemem || return 1
+        fi
+
+    elif [[ $NODEMEM -gt $free_to_use ]]; then
+        echo "  ERROR: Requested node memory setting is larger than usable free memory: ${NODEMEM} MB > ${free_to_use} MB (free ${total_free} MB - buffer ${mem_buffer} MB). Abort!"
+        exit 1
+    fi
+}
+
+calc_nodemem
 
 export NODE_OPTIONS=--max-old-space-size=${NODEMEM}
 
 echo "Setting Node Options:"
 env | grep NODE
 
 if [[ $(uname -m) == armv6l ]]; then
-  echo "  You are running on a hardware with less resources. Building
-  the webapp might fail. If so, try to install the stable
-  release installation instead."
+    echo "
+-----------------------------------------------------------
+| You are running a hardware with limited resources.      |
+| Building the Web App takes significantly more time      |
+| and might fail.                                         |
+-----------------------------------------------------------
```

Comment:
What about unifying this with `show_slow_hardware_message` ?

Reply:
Did think about this too.
Which message should we use? 
And do you want to import the helper script or just duplicat the message?

Reply:
Let's not merge it and instead give a little bit more support

```
Building the Web App takes significantly more time. In case it fails, check the documentation to trouble shoot.
```

Reply:
You don't have to write a second time, that the hardware is slow ;-)